### PR TITLE
#3611  JSON Schema package

### LINKED ISSUE #3611: Add JSONSchema form package
## Is your feature request related to a problem? Please describe.

We want to create dynamically created forms from a JSON schema. There are a lot of packages which can do this:

One of the most widely used is https://github.com/rjsf-team/react-jsonschema-form however this is not compatible with react-native.

There are a couple of forks/themese being maintained which support react-native:
- https://www.npmjs.com/package/react-native-web-jsonschema-form - primarily for web and we have to install a bunch of extra dependencies (mainly expo ones) to make it work.
- https://github.com/royaizenberg/react-native-jsonschema-form - pretty abandoned
- https://github.com/zhigang1992/react-native-jsonschema-form - looks pretty promising, but doesn't theme/support the full set of widgets that the original project does, so will still require a decent amount of implementing different components.

## Describe the solution you'd like

Either
- Utilize zhigang1992's version and extend it
	- Pro: Potentially *slgihtly* less code to write
	- Con: We have to rewrite most of the widgets so it really doesn't offer *that* much
	- Con: We are extending an extension of a project, could have nightmares later when/if it's abandoned
- Create our own theme/extension of `react-schema-form`
	- Pro: Saving a decent amount of time by having a package organise which components should be rendered given a schema, out of the box.
	- Con: Potentially losing a decent amount of time by having a package organise which components should be rendered given a schema, out of the box - huh?
	- Pro: full customized widgets is good for us.
	- Con/Pro(??): Might need to do some commits into the mainline branch to ensure the full project is RN compatible.
- Create our own JSON schema form generator
	- if only we had the time 🤣 

Personally I think making our own theme/extension for `react-schema-form` is a good middle ground. The project has 10k+ stars and 60k downloads/week so it is a solid project unlikely to be sunset. They have issues and seem keen on adding RN support if anyone offered a PR which we could do later, so that someone else could decide they don't like our theme and not use it.

## Implementation

Add the package if any to the code base to start utilizing

## Describe alternatives you've considered

Above
## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3611 

## Change summary

- Adds `rjsf/core`
- Adds some placeholder components to be overriden. Hopefully giving some structure to future work.
- Added comments and hopefully a solid base structure to move forward with

## Testing

Internal

### Related areas to think about

N/A

### CODE REVIEW

**Comment by joshxg** [comment]
image

Just made it like this, where we can start updating each of the templates/fields/widgets that we want to implement afterward

File: src/widgets/JSONForm/JSONForm.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            }}
+            // eslint-disable-next-line no-console
+            onSubmit={form => console.log('onSubmit:', form)}
+            ref={formRef}
+            schema={exampleSchema}
+          >
+            {children ?? (
+              <PageButton
+                onPress={e => {
+                  // eslint-disable-next-line no-console
```
Comment (mark-prins) [inline src/widgets/JSONForm/JSONForm.js:210]: :just a minor comment.. this hint isn't needed

File: src/widgets/JSONForm/fields/Description.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable react/prop-types */
```
Comment (mark-prins) [inline src/widgets/JSONForm/fields/Description.js:1]: strictly speaking.. not necessary. but will be, I expect

**Review by mark-prins — APPROVED** [review]
very nice!
I was unsure about the need for `fields` `templates` and `widgets` .. will see how it looks when implemented. I'm sure there's good reasons which will become clearer.

Will need to double check the eslint hinting when implemented, not all is needed atm; though I think will be

**Pull request merged**